examples/CtxtFilterExample.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright Schrodinger, LLC
+ */
+
+"use strict";
+
+const { AvatarCell, TextCell } = require('./helpers/CtxtCells');
+const { DataCtxt } = require('./helpers/HOC');
+const FakeObjectDataListStore = require('./helpers/FakeObjectDataListStore');
+const { Table, Column, Cell } = require('fixed-data-table-2');
+const React = require('react');
+
+class DataListWrapper {
+  constructor(data) {
+    this._data = data;
+    this._indexMap = null;
+    this._callback = null;
+  }
+
+  // The callback is used for triggering re-rendering
+  setCallback(cb) {
+    this._callback = cb;
+  }
+
+  setIndexMap(index) {
+    this._indexMap = index;
+    this._callback();
+  }
+
+  getSize() {
+    if (this._indexMap === null) {
+      return this._data.getSize();
+    }
+
+    return this._indexMap.length;
+  }
+
+  getObjectAt(index) {
+    if (this._indexMap === null) {
+      return this._data.getObjectAt(index);
+    }
+
+    return this._data.getObjectAt(
+      this._indexMap[index]
+    );
+  }
+}
+
+const DataTable = DataCtxt(Table);
+
+class FilterTable extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { data, filters, ...other } = props;
+    this.state = {
+      rawData: data,
+      filteredData: new DataListWrapper(props.data),
+      filters: Object.assign({}, filters),
+      other
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (JSON.stringify(nextProps.filters) !== JSON.stringify(this.state.filters)){
+      this.setState({
```

Comment:
Since the prop is an object the values are passed as a reference and without the cloning I could not detect when the filter changed and when to update the parameter.